src/client/builtin_scene/materials/web_content_instanced.cpp

```diff
@@ -61,13 +83,40 @@ namespace builtin_scene::materials
     auto glContext = glContext_.lock();
     assert(glContext != nullptr);
 
+    // Set up border data callback for instanced meshes
+    if (mesh->isInstancedMesh() && borderDataTexture_ && borderDataTexture_->isInitialized())
+    {
+      auto& instancedMesh = mesh->getHandleCheckedAsRef<InstancedMeshBase>();
+      instancedMesh.setBorderDataUpdateCallback([this](const std::vector<std::shared_ptr<Instance>>& instances) {
```

Comment:
You don't need a callback for updating the underlay texture. At `SceneRenderer::drawInstancedMeshImpl()`, it must have a `RenderableInstancesList& instances` to draw, at that time, you can try to upload the border data of these instances to the texture data.